### PR TITLE
Feature: Improve music videos

### PR DESCRIPTION
**Pull request by wutschel**
## Description

1. Improve displaying music videos:
a) Show artist information
b) Add support to sort by artist and genre
c) Sort by "Title" instead of "Name"
Screenshot: https://abload.de/img/bildschirmfoto2021-09r2k6q.png

2. While working on this I also improved displaying runtime. With this PR runtime string are kept empty (`""`), if the `runtime == 0`.
Screenshot: https://abload.de/img/bildschirmfoto2021-091kk4u.png

3. Another improvement done while working on the music videos sort-by enhancement is more robustness in the sort-algorithm which now processes `NSArray` as well.

## Summary for release notes

Feature: Improve displaying music videos
Improvement: Improve sorting robustness
Improvement: Improve displaying runtime

### CODE REVIEW

File: XBMC Remote/Utilities.m [hunk 1]
```diff
@@ -615,11 +615,8 @@ + (NSString*)getTimeFromDictionary:(NSDictionary*)dict key:(NSString*)key sec2mi
     else if ([value isKindOfClass:[NSArray class]]) {
         runtime = [NSString stringWithFormat:@"%@", [value componentsJoinedByString:@" / "]];
     }
-    else if ([value intValue]) {
-        runtime = [NSString stringWithFormat:@"%d min", [value intValue]/secondsToMinute];
-    }
-    else {
-        runtime = [NSString stringWithFormat:@"%@", value];
+    else if ([value intValue] / secondsToMinute) {
+        runtime = [NSString stringWithFormat:@"%d min", [value intValue] / secondsToMinute];
```
Comment (kambala-decapitator) [inline XBMC Remote/Utilities.m:619]: please avoid copy-paste

File: XBMC Remote/DetailViewController.m [hunk 2]
```diff
@@ -2278,6 +2278,10 @@ - (UITableViewCell*)tableView:(UITableView*)tableView cellForRowAtIndexPath:(NSI
             [genre setMinimumScaleFactor:10.0/11.0];
             [genre sizeToFit];
         }
+        else if ([item[@"family"] isEqualToString:@"musicvideoid"]) {
+            genre.text = [Utilities getStringFromDictionary:item key:@"genre" emptyString:@""];
```
Comment (kambala-decapitator) [inline XBMC Remote/DetailViewController.m:2282]: maybe makes sense to add a method overload that passes `emptyString:@""` automatically?

**Review by kambala-decapitator — CHANGES_REQUESTED** [review]

File: XBMC Remote/DetailViewController.m (hunk 2 above)
Comment (wutschel) [reply XBMC Remote/DetailViewController.m:2282]: There is a balanced amount of calls with `emptyString:@""` and `emptyString:@"-"`. I would like to keep clarity on what `emptyString` is from the call itself. Hoping this is fine for you.

File: XBMC Remote/Utilities.m (hunk 1 above)
Comment (wutschel) [reply XBMC Remote/Utilities.m:619]: The alternative in my opinion is harder to read:

```
else {
    int minutes = [value intValue] / secondsToMinute;
    runtime = minutes ? [NSString stringWithFormat:@"%d min", minutes] : runtime;
}
```

File: XBMC Remote/DetailViewController.m (hunk 2 above)
Comment (kambala-decapitator) [reply XBMC Remote/DetailViewController.m:2282]: yep fine, thanks

File: XBMC Remote/Utilities.m (hunk 1 above)
Comment (kambala-decapitator) [reply XBMC Remote/Utilities.m:619]: but the intent is much clearer. Besides, you don't have an `else` case any more.

Comment (wutschel) [reply XBMC Remote/Utilities.m]: I do not have a strong opinion on this, so I applied the change and already squashed it (first commit).

**Pull request merged**